ts/src/pro/woofipro.ts

```diff
@@ -0,0 +1,1150 @@
+                delete client.subscriptions['authenticated'];
+            }
+        }
+    }
+
+    async authenticate (params = {}) {
+        this.checkRequiredCredentials ();
+        const url = this.urls['api']['ws']['private'] + '/' + this.accountId;
+        const client = this.client (url);
+        const messageHash = 'authenticated';
+        const event = 'auth';
+        const future = client.future (messageHash);
+        const authenticated = this.safeValue (client.subscriptions, messageHash);
+        if (authenticated === undefined) {
+            const ts = this.nonce ().toString ();
+            const auth = ts;
+            let secret = this.secret;
+            if (secret.indexOf ('ed25519:') >= 0) {
+                const parts = secret.split ('ed25519:');
+                secret = parts[1];
+            }
+            const signature = eddsa (this.encode (auth), this.base58ToBinary (secret), ed25519);
+            const request = {
+                'event': event,
+                'params': {
+                    'orderly_key': this.apiKey,
+                    'sign': signature,
+                    'timestamp': ts,
+                },
+            };
+            const message = this.extend (request, params);
+            this.watch (url, messageHash, message, messageHash);
+        }
+        return await future;
+    }
+
+    async watchPrivate (messageHash, message, params = {}) {
+        await this.authenticate (params);
+        const url = this.urls['api']['ws']['private'] + '/' + this.accountId;
+        const requestId = this.requestId (url);
+        const subscribe = {
+            'id': requestId,
+        };
+        const request = this.extend (subscribe, message);
+        return await this.watch (url, messageHash, request, messageHash, subscribe);
+    }
+
+    async watchOrders (symbol: Str = undefined, since: Int = undefined, limit: Int = undefined, params = {}): Promise<Order[]> {
+        /**
+         * @method
+         * @name woofipro#watchOrders
+         * @description watches information on multiple orders made by the user
+         * @see https://orderly.network/docs/build-on-evm/evm-api/websocket-api/private/execution-report
+         * @see https://orderly.network/docs/build-on-evm/evm-api/websocket-api/private/algo-execution-report
+         * @param {string} symbol unified market symbol of the market orders were made in
+         * @param {int} [since] the earliest time in ms to fetch orders for
+         * @param {int} [limit] the maximum number of order structures to retrieve
+         * @param {object} [params] extra parameters specific to the exchange API endpoint
+         * @param {bool} [params.stop] true if stop order
+         * @returns {object[]} a list of [order structures]{@link https://docs.ccxt.com/#/?id=order-structure}
+         */
+        await this.loadMarkets ();
+        const stop = this.safeBool (params, 'stop', false);
```

Comment:
the unified parameter is `trigger` now we should read `stop` as well but add only `trigger` to the documents